samza-core/src/main/java/org/apache/samza/clustermanager/ContainerManager.java

```diff
@@ -62,33 +83,67 @@ public ContainerManager(SamzaApplicationState samzaApplicationState, ClusterReso
   }
 
   /**
-   * Handles the container start action for both active & standby containers.
+   * Handles the container start action for both active & standby containers. This method is invoked by the allocator thread
+   *
+   * Case 1. If the container launch request is due to an existing container placement action, issue a stop on active
+   *         container & wait for the active container to be stopped before issuing a start.
+   * Case 2. If StandbyContainer is present refer to {@code StandbyContainerManager#checkStandbyConstraintsAndRunStreamProcessor}
+   * Case 3. Otherwise just invoke a container start on the allocated resource for the pending request
+   *
+   * TODO: SAMZA-2399: Investigate & configure a timeout for container stop if needed
    *
    * @param request pending request for the preferred host
    * @param preferredHost preferred host to start the container
    * @param allocatedResource resource allocated from {@link ClusterResourceManager}
    * @param resourceRequestState state of request in {@link ContainerAllocator}
    * @param allocator to request resources from @{@link ClusterResourceManager}
+   *
+   * @return true if the allocator thread supposed to sleep for sometime before checking the stop of active container
+   *         for an undergoing container placement action, false otherwise
```

Comment:
This return semantics is kinda confusing.
Could this method instead, 
return true if the launch is complete, false otherwise?

The caller can then use the fact that launch is not-complete and check again (however it wants).